### ISSUE TITLE
Add "local time - Day of the Month" as a possible dimension in Custom Report

### ISSUE THREAD

**Issue by chulinguy** [piwik/piwik#20950]
## Summary
We previously encountered an issue that the same set of data would look different in Admin UI vs Custom Report

1) In Admin UI, we can do a segment by website time, let's say a week worth of data, then the we would see 7 different data points (CORRECT)

2) In Custom Reports, we can do a segment by website time, but since "Day of the Month" is only available in UTC time, we would get a week worth of data spread across 8 different data points (INCORRECT)

We've mitigated this issue temporarily by changing our website time to be UTC to avoid the whole 7-vs-8-data-point problem.
However, since our client still assumes the data we're providing data in American time, we're technically providing incorrect data to them.  

Therefore, I would like to request to have "local time - Day of the Month" as a possible dimension in Custom Report

**Comment by michalkleiner** [comment]
Hi @chulinguy, 
thank you for reaching out. This Github repository is for development issues and improvements relating to the core Matomo application, so for support queries and feature requests on the Custom Reports premium plugin we'd recommend to contact the dedicated support team who will put the suggested enhancement to the product owner for prioritisation. Appreciate your understanding!

**Comment by michalkleiner** [comment]
Apologies @chulinguy, I learnt that the dimensions are provided by the core itself, so opening the enhancement request makes sense here. The team will discuss this with the product owner and prioritise accordingly.

**Comment by michalkleiner** [comment]
@sgiehl do you think this could even be seen as a bug since Day of month is not respecting the timezone of the website in custom reports? Would we need a new dimension and leave the existing one as the UTC one?

**Comment by sgiehl** [comment]
I guess he was referring to this dimension: https://github.com/matomo-org/matomo/blob/5.x-dev/plugins/CoreHome/Columns/VisitLastActionDayOfMonth.php#L22
It's name actually says that it is in UTC, so that should be fine. 

I had a closer look into this and adding that for the "local time" might actually not possible that easily. We store the local time as a dimension. But this only covers the time and not the date. 
Note: the local time isn't the timezone configured for the website. It's the time the user viewing the website has. So that can be from any timezone. So without storing the local date, we might not be able to provide this.

But guess the request here is more likely for the "site time". Which could be solved easily by duplicating and adjusting e.g. this dimension: https://github.com/matomo-org/matomo/blob/5.x-dev/plugins/CoreHome/Columns/ServerMinute.php#L18

**Comment by chulinguy** [comment]
@sgiehl I was not aware of the nuisance between "local time" vs "site time". 
In our case, we're interested in site time.  

Based on your comment, it would be easy to add  "site time - day of the month" as a dimension?

**Comment by sgiehl** [comment]
> Based on your comment, it would be easy to add "site time - day of the month" as a dimension?

Yes, should be quite easy. But we might also need to add some tests for it, to prove it works as expected.
Our product team will decide when we will work on this improvement.

**Comment by chulinguy** [comment]
Hi @michalkleiner, it's been two months.  Do we have an ETA for when this easy change would be implemented?

**Comment by michalkleiner** [comment]
Hi @chulinguy, we have this enhancement in the backlog and the product owner will decide when this will be worked on. If you are keen to open a pull request adding the site time dimension as indicated by sgiel before, we will definitely review it. Thank you for understanding.

**Comment by chulinguy** [comment]
Is there a way to escalate this? It's been 4 months and literally we have not been able to provide correct data to our clients.

**Comment by Stan-vw** [comment]
Thank you for following up. We have it planned a few sprints from now. It's difficult to give an exact timeline as we're also working to get a stable version of Matomo 5 out at the moment.

**Comment by chulinguy** [comment]
Hi, it's a new year.  Do we have updated insight for when this feature will be ready?

**Comment by goochj03** [comment]
Another request to see a local date metric added to custom reports in an easily understandable format. e.g. Friday, June 14, 2024 - 15:22:23
